src/container/srv_container.c

```diff
@@ -1853,59 +1907,64 @@ cont_agg_eph_leader_ult(void *arg)
 			goto yield;
 		}
 
-		d_list_for_each_entry_safe(ec_agg, tmp, &svc->cs_ec_agg_list, ea_list) {
-			daos_epoch_t min_eph = DAOS_EPOCH_MAX;
+		d_list_for_each_entry_safe(eph_ldr, tmp, &svc->cs_cont_ephs_leader_list, ea_list) {
+			daos_epoch_t min_ec_agg_eph = DAOS_EPOCH_MAX;
+			daos_epoch_t min_stable_eph = DAOS_EPOCH_MAX;
 			int	     i;
 
-			if (ec_agg->ea_deleted) {
-				d_list_del(&ec_agg->ea_list);
-				D_FREE(ec_agg->ea_server_ephs);
-				D_FREE(ec_agg);
+			if (eph_ldr->ea_deleted) {
+				d_list_del(&eph_ldr->ea_list);
+				D_FREE(eph_ldr->ea_server_ephs);
+				D_FREE(eph_ldr);
 				continue;
 			}
 
-			for (i = 0; i < ec_agg->ea_servers_num; i++) {
-				d_rank_t rank = ec_agg->ea_server_ephs[i].rank;
+			for (i = 0; i < eph_ldr->ea_servers_num; i++) {
+				d_rank_t rank = eph_ldr->ea_server_ephs[i].re_rank;
 
 				if (d_rank_in_rank_list(&fail_ranks, rank)) {
 					D_DEBUG(DB_MD, DF_CONT" skip %u\n",
 						DP_CONT(svc->cs_pool_uuid,
-							ec_agg->ea_cont_uuid),
+							eph_ldr->ea_cont_uuid),
 						rank);
 					continue;
 				}
 
-				if (ec_agg->ea_server_ephs[i].eph < min_eph)
-					min_eph = ec_agg->ea_server_ephs[i].eph;
+				if (eph_ldr->ea_server_ephs[i].re_ec_agg_eph < min_ec_agg_eph)
+					min_ec_agg_eph = eph_ldr->ea_server_ephs[i].re_ec_agg_eph;
+				if (eph_ldr->ea_server_ephs[i].re_stable_eph < min_stable_eph)
+					min_stable_eph = eph_ldr->ea_server_ephs[i].re_stable_eph;
 			}
 
-			if (min_eph == ec_agg->ea_current_eph)
+			if (min_ec_agg_eph == eph_ldr->ea_current_ec_agg_eph &&
+			    min_stable_eph == eph_ldr->ea_current_stable_eph)
 				continue;
 
 			/**
 			 * NB: during extending or reintegration, the new
 			 * server might cause the minimum epoch is less than
-			 * ea_current_eph.
+			 * ea_current_ec_agg_eph.
 			 */
-			D_DEBUG(DB_MD, DF_CONT" minimum "DF_U64" current "DF_X64"\n",
-				DP_CONT(svc->cs_pool_uuid, ec_agg->ea_cont_uuid),
-				min_eph, ec_agg->ea_current_eph);
-
-			cur_eph = d_hlc2sec(ec_agg->ea_current_eph);
-			new_eph = d_hlc2sec(min_eph);
+			D_DEBUG(DB_MD, DF_CONT" min_ec_agg_eph "DF_X64" current "DF_X64
+				", min_stable_eph "DF_X64" current "DF_X64".\n",
+				DP_CONT(svc->cs_pool_uuid, eph_ldr->ea_cont_uuid),
+				min_ec_agg_eph, eph_ldr->ea_current_ec_agg_eph,
+				min_stable_eph, eph_ldr->ea_current_stable_eph);
+
+			cur_eph = d_hlc2sec(eph_ldr->ea_current_ec_agg_eph);
+			new_eph = d_hlc2sec(min_ec_agg_eph);
```

Comment:
Does it only affect ec_agg_epoch or also stable_epoch?

Reply:
this is to throw a warning if the EC agg epoch boundary is updated too slow (> 600S).

Both ec_agg_epoch and stable_epoch are considered by other code in this func